sdk/resourcemanager/docs/THREAD_POOL_AND_CONNECTION_POOL.md

```diff
@@ -0,0 +1,90 @@
+		.build())
+```
+
+### <a name="httpthreadpool"></a>Configure thread pool for the singleton HttpClient:
+By default the TCP client uses an “Event Loop Group”, where the number of the worker threads equals the number of processors available to the runtime on initialization (but with a minimum value of 4). When you need a different configuration, you can use one of the LoopResource#create methods.
+```java
+// Thread pool configuration.  
+// Official Reactor Netty documentation for defaults: https://projectreactor.io/docs/netty/release/reference/#client-tcp-level-configurations-event-loop-group  
+singletonHttpClientBuilder 
+	.eventLoopGroup(LoopResources  
+	    .create(  
+            "client-thread-pool", // thread pool name  
+		    Runtime.getRuntime().availableProcessors() * 2, // thread pool size  
+		    true)  
+         // we use our custom event loop here, disable the native one  
+	    .onClient(false))  
+    .build();
+```
+
+### <a name="azureidentity"></a>Configure Azure Identity client to use the singleton HttpClient (DefaultAzureCredential for example):
+By default, Azure Identity uses `ForkJoinPool.commonPool()` for token acquisition. The pool size equals the number of processors available to the runtime on initialization minus 1 (with a minimum of 1). It's a singleton pool and is shared by default. Default pool configuration should be sufficient for most use cases.
+```java
+HttpClient singletonHttpClient = singletonHttpClientBuilder.build();
+// Use the singleton httpClient for Azure Identity
+final AzureProfile profile = new AzureProfile(AzureEnvironment.AZURE);  
+final TokenCredential credential = new DefaultAzureCredentialBuilder()  
+    .authorityHost(profile.getEnvironment().getActiveDirectoryEndpoint())  
+    .executorService(ForkJoinPool.commonPool()) // thread pool for executing token acquisition, usually we leave it default  
+    .httpClient(singletonHttpClient)
+  .build();
+```
+
+### <a name="azureclient"></a>Configure Azure Clients to use the singleton HttpClient and Azure Identity client:
+```java
+// Use the singleton httpClient for your Azure client
+AzureResourceManager azureResourceManager = AzureResourceManager  
+	.configure()  
+    .withLogLevel(HttpLogDetailLevel.BASIC)  
+    .withHttpClient(singletonHttpClient)
+    .authenticate(credential, profile)  
+    .withSubscription(subscriptionId); // your subscription ID, can be different for different Azure clients
+```
+
+### <a name="reactor"></a>Configure Reactor thread pool size:
+By default, Azure Client subscribes on Schedulers.parallel(), which has a thread pool of size equal to available processor count. It is a static pool and is shared in nature.
+You can change the pool size by specifying the variable `reactor.schedulers.defaultPoolSize` in environment variables, whether through command line arguments, or System.setProperty.
+```java
+System.setProperty("reactor.schedulers.defaultPoolSize", NettyRuntime.availableProcessors() + "");
```

Comment:
`String.valueOf` might be better?

Also does it (if via code) need to appear before any Reactor to take effect? I kind of assume it does not have effect after Reactor internalized `parallel` pool.

Reply:
I think it should be before class `Schedulers` get loaded by classloader, which means before `Schedulers` get referenced in code. https://github.com/reactor/reactor-core/blob/2158822d9a68ca15fa9f2660f56fa2f7ad161a43/reactor-core/src/main/java/reactor/core/scheduler/Schedulers.java#L69-L80

Reply:
My point is, if user could get confused about whether calling System.setProperty can take effect or not, we probably don't make this sample in the first place. Just say it can be configured in env var should be enough.